framework/doc/content/source/meshgenerators/TransfiniteMeshGenerator.md

```diff
@@ -0,0 +1,59 @@
+# TransfiniteMeshGenerator
+
+The `TransfiniteMeshGenerator` produces two-dimensional meshes from a set of 4 corner vertices by interpolating the edges into the inner domain. The resulting mesh  consists of QUAD4 elements of aspect ratio and skewness determined by the points on the boundary edges.
+
+!syntax description /Mesh/TransfiniteMeshGenerator
+
+## Overview
+
+!media media/meshgenerators/quad_convention.png style=float:right;width:20%;margin-left:2%; caption=Fig. 1: Convention for the corner vertices order.
+
+Given a set of 4 corners the TransfiniteMeshGenerator constructs by default the straight edges that connect the vertices and fills up the interior with straight lines. The test, [quadrilater_generator.i](test/tests/meshgenerators/transfinite_generator/quadrilater_generator.i),
+illustrates the use of the `TransfiniteMeshGenerator` object to construct general quadrilaterals as in Fig. 2. By default the distribution of points along edges is considered equidistant, unless specified otherwise by the user using the parameters `bias_x` or `bias_y`. The core of the algorithm interpolates/fills up the inner domain for any edge type, curved, parametrized, discrete. The current implementation considers all these cases with special attention given to arcs of a circle which are widely encountered in scientific computing applications. 
+
+The convention for the corners numbering should follow Fig. 1, prescribed in the order +(0,0) - (0,1) - (1,1) - (1,0)+. The direction along edges, important if the user provides them using the DISCRETE option, should follow the direction of the arrows as in Fig. 1.
```

Comment:
I've noticed in my use of this capability that the winding order (as measured by the test object auxkernel WindingOrder, see test at test/tests/meshgenerators/concentric_circle_mesh_generator/concentric_circle_mesh2.i ) varies with the order in which the nodes are prescribed.  The given order creates a negative winding order (as desired) while the reverse, incorrect error gives a positive winding error. Is there a way to add in a warning in the code or a note in the documentation page here to help out bad-at-reading-users like me please?

Reply:
Oh I need to hear more about this, I actually had an issue with the biases that I couldn't find a good case to see it fail properly till recently, and I had to fix a typo that was reversing the sign. Could you please let me know if you have this issue also pulling the latest version?

Reply:
(put this in the wrong spot, sorry about the duplication) As promised, the MWE/MNWE with the winding order aux kernel. As you'll see, it is all user error (between my keyboard and my chair) that caused the issue, and I'm not sure if it is possible to guard against this sort of user error in the code:
```
[Mesh]
  [quadrilateral]
    type = TransfiniteMeshGenerator
    #Gives a positive winding order which is not good for extruding to 3D
    corners = '-0.6 -0.5 0.0
                0.8 -0.5 0.0
                0.7  0.5 0
               -0.6  0.5 0'

    #Gives a negative winding order and works well for extruding to 3D
    # corners = '-0.6  0.5 0
    #             0.7  0.5 0
    #             0.8 -0.5 0.0
    #            -0.6 -0.5 0.0'
    nx = 4
    ny = 9
    bottom_type = LINE
    top_type = LINE
    right_type = LINE
    left_type = LINE
  []
[]

[AuxVariables]
  [winding_order]
    order = CONSTANT
    family = MONOMIAL
  []
[]

[AuxKernels]
  [winding_order]
    type = WindingOrder
    variable = winding_order
  []
[]

[Variables]
  [u]
  []
[]

[Kernels]
  [diff]
    type = Diffusion
    variable = u
  []
[]

[BCs]
  [left]
    type = DirichletBC
    variable = u
    boundary = left
    value = 0
  []
  [right]
    type = DirichletBC
    variable = u
    boundary = right
    value = 1
  []
[]

[Postprocessors]
  [min_winding_order]
    type = ElementExtremeValue
    value_type = min
    variable = winding_order
  []
  [max_winding_order]
    type = ElementExtremeValue
    value_type = max
    variable = winding_order
  []
[]

[Executioner]
  type = Steady
[]

[Outputs]
  exodus = true
  csv = true
  hide = winding_order
[]
```

Reply:
@oanaoana 
The API call to fix the winding order
```  MeshTools::Modification::orient_elements(*mesh);```

Reply:
@sapitts : In general... in libMesh / MOOSE we always use a "right hand rule" orientation.  That is: a counter-clockwise "winding" will have a positive (pointing at you from the surface) normal.

Reply:
Please mark this conversation also as resolved, in light of the libmesh update.